### DISCUSSION TITLE
Switch to a layout that's not in the layouts list

### DISCUSSION THREAD

**Discussion by 80KiloMett** [qtile/qtile#2385] [Q&A]
I normally change layouts just by cycling through the layouts list. That's why I don't want to clutter it with too many layouts. Is there a way to load a layout that's not in the list?

Right now Max, Cloumn and MonadTall are in my list and I rarely feel the need for anything else. Today however I found a specific usecase for the bsp layout. So I would like to be able to load it when I feel the need without adding it to my layouts cycle.

**Comment by ramnes** [comment]
It's not something natively supported AFAIK, but I feel it's exactly the kind of niche use case that Qtile is made for by being configured in Python. You could write and bind custom functions in your configuration for this, one to cycle through your main layouts and one to switch to BSP.

**Reply by 80KiloMett** [reply]
Yeah, I've already tried hacking around with the InteractiveCommandClient, also to get some introspection. The only way I've found so far is something like this:
```
c = InteractiveCommandClient()
c.group.eval("from libqtile.layout import Bsp\nself.layouts.append(Bsp())")
```
Pretty hacky. And it doesn't work that well. Windows that were on screen before injecting the layout stuck to the original layout.

Before that I tried to build a custom list type that has a builtin dict and a __getattr__ method, that distiquishes between strings and numbers. I would then import that custom list type into my config.py and build my layers list using that type. I then wanted to use `lazy.to_layouts_index("bsp")` to access the layout in the dictionary.
Unfortunatly qtile doesn't seem to just reference the layers list in the config.py but rather copy its elements and this approach would have failed anyway due to range checking on the index.

I feel like I tried out the hackiest approaches first. I'd be happy if someone could maybe point me in the right direction and I'll take it from there.

P.S. Sorry for posting this on the issue tracker originally. I'll stick to discussions with questions like this in the future. ;)
